dds_web/api/user.py

```diff
@@ -452,8 +447,29 @@ def post(self):
             raise ddserr.DDSArgumentError(message="User is already in desired state!")
 
         # TODO: Check if user has lost access to any projects and if so, grant access again.
+        if action == "reactivate":
+            user.active = True
+
+            # TODO: Super admins (current_user) don't have access to projects currently, how handle this?
+            list_of_projects = None
+            if user.role in ["Project Owner", "Researcher"]:
+                list_of_projects = [x.project for x in user.project_associations]
+            elif user.role in ["Unit Personnel", "Unit Admin"]:
+                list_of_projects = user.unit.projects
+
+            for proj in list_of_projects:
+                if proj.is_active:
+                    project_keys_row = models.ProjectUserKeys.query.filter_by(
+                        project_id=proj.id, user_id=user.username
+                    ).one_or_none()
+                    if not project_keys_row:
+                        share_project_private_key(
+                            from_user=current_user, to_another=user, project=proj
+                        )
```

Comment:
I believe you can copy paste this into the #885 no?

Reply:
@alneberg It's very possible that we can use this or extremely similar code in #885, but I decided to just make the skeleton in that first of all. The person who works on the `ProjectAccess` endpoint could potentially create a new function in that class so that we can just call it from this function too.

Reply:
I can add a comment in the next card for this so that noone re-invents the wheel